lib/rules/no-array-constructor.js

```diff
@@ -37,11 +79,38 @@ module.exports = {
          */
         function check(node) {
             if (
-                node.arguments.length !== 1 &&
-                node.callee.type === "Identifier" &&
-                node.callee.name === "Array"
-            ) {
-                context.report({ node, messageId: "preferLiteral" });
+                node.callee.type !== "Identifier" ||
+                node.callee.name !== "Array" ||
+                node.arguments.length === 1 &&
+                node.arguments[0].type !== "SpreadElement") {
+                return;
+            }
+
+            const variable = getVariableByName(sourceCode.getScope(node), "Array");
+
+            if (variable && variable.identifiers.length === 0) {
+                const argsText = getArgumentsText(node);
+                let fixText;
+                let messageId;
+
+                if (isStartOfExpressionStatement(node) && needsPrecedingSemicolon(sourceCode, node)) {
```

Comment:
Same here. Just a short explanation of what we are checking and why.

Reply:
All done in https://github.com/eslint/eslint/pull/17711/commits/c99888efa52ad4a257d0218fd78e820c88fa5962. Thanks for the suggestions, I should learn to use comments more than I'm used to.

Reply:
Thanks! Yeah, I'm being more specific about comments in PRs now after spending weeks in the code path analysis code where there were very few comments. Trying to save future-us more pain.